src/librustdoc/html/render/mod.rs

```diff
@@ -1031,11 +1031,11 @@ themePicker.onblur = handleThemeButtonsBlur;
     // with rustdoc running in parallel.
     all_indexes.sort();
     {
-        let mut v = String::from("var searchIndex = JSON.parse('{\\\n");
+        let mut v = String::from("var searchIndex = JSON.parse(\"{\\n");
         v.push_str(&all_indexes.join(",\\\n"));
         // "addSearchOptions" has to be called first so the crate filtering can be set before the
         // search might start (if it's set into the URL for example).
-        v.push_str("\\\n}');\naddSearchOptions(searchIndex);initSearch(searchIndex);");
+        v.push_str("\\\n}\");\naddSearchOptions(searchIndex);initSearch(searchIndex);");
```

Comment:
```suggestion
        v.push_str("\\\n}');\naddSearchOptions(searchIndex);initSearch(searchIndex);");
```

Again, this needs to stay using a single quote for now.